lib/maths/common/CTools.cc

```diff
@@ -774,7 +774,7 @@ operator()(const CLogTDistribution& logt, double x, maths_t::ETail& tail) const
     double step = std::max(b2, std::exp(l) - b2);
     double growthFactor = 1.0;
     for (;;) {
-        if (maxIterations == 0 || f2 <= fx) {
+        if (maxIterations == 2 || f2 <= fx) {
```

Comment:
It's a bit weird to stop at 2.  I guess it relates to the way `step` is handled differently when it's a certain proportion of `MAX_ITERATIONS`, but please could you add a comment here to confirm it's not a mistake.

Reply:
Actually, this was accidentally committed now I added protection for maxIterations being 0 in the solver code.

Reply:
See [33a3ef2](https://github.com/elastic/ml-cpp/pull/2197/commits/33a3ef2464aab0cb830532f243d7a5e7bcb263b5).